### ISSUE TITLE
Unknown Send type in generator with yield from

### ISSUE THREAD

**Issue by AlanBogarin** [microsoft/pyright#10279]
## Environment data

-   Pylance version: 2025.4.1
-   OS and version: Windows 11
-   Python version (& distribution if applicable, e.g. Anaconda): 3.13.2

## Code Snippet

```python
from collections.abc import Generator


def random_generator() -> Generator[int, str, bool]:
    for _ in range(3):
        x = yield int()
        assert isinstance(x, str), f"str expected, got {x}"
    return bool()

# reportUnknownParameterType: Return type, "Generator[int, Unknown, bool]", is partially unknown
def yieldfrom_func():
    return (yield from random_generator())
```

## Repro Steps

1. Enable type checking mode: `strict`

## Expected behavior

I expected `yieldfrom_func` to assume the type `Generator[int, str, bool]` based on the `yield from` statement, which references `Generator[int, str, bool]`. Why? Because of how it works. `yield from` causes all yields from the target generator to be used as if they had been called from the original generator.
Here is some code to analyze the processes occurring inside a generator with `yield from`.

```python
from collections.abc import Callable
from typing import Any, Generator, cast


def log(func: Callable[..., Any], msg: str) -> None:
    print(f"{func.__name__}: {msg}")


def consume[Y, S, R](
    genfunc: Callable[[], Generator[Y, S, R]], yield_dtype: type[Y], *sends: S
) -> R:
    generator = genfunc()
    n = -1
    try:
        while True:
            if n == -1:
                yield_value = next(generator)
                log(genfunc, f"next() --> {yield_value!r}")
            else:
                send_value = sends[n]
                yield_value = generator.send(send_value)
                log(genfunc, f"send({send_value!r}) --> {yield_value!r}")
            assert isinstance(
                yield_value, yield_dtype
            ), f"dtype is {yield_dtype.__name__}, {genfunc.__name__!r} yields {yield_value!r}"
            n += 1
    except IndexError:
        raise ValueError("Unrealistic for iteration")
    except StopIteration as e:
        return_value = cast(R, e.args[0])
    log(genfunc, f"return {return_value}\n")
    return return_value


def random_generator() -> Generator[int, str, bool]:
    for _ in range(3):
        x = yield int()
        assert isinstance(x, str), f"str expected, got {x}"
    return bool()


# Pylance shows: Generator[int, Unknown, bool]
def yieldfrom_func() -> Generator[int, str, bool]:
    return (yield from random_generator())


sends = ("", "", "")
origin = consume(random_generator, int, *sends)
yieldfrom = consume(yieldfrom_func, int, *sends)

assert origin == yieldfrom, f"{origin=}, {yieldfrom=}"
```

## Actual behavior

Pylance assumes that the function `yieldfrom_func` returns the type `Generator[int, Unknown, bool]`

![Image](https://github.com/user-attachments/assets/5c02aef7-c022-441e-bc7e-e6caab72f42b)

## Logs

<details>

<summary>VSCode Pylance Logs</summary>

```
[Info] (14400) Execution environment: 3.13.2 (global)
[Info] (14400)   Extra paths:
[Info] (14400)     (none)
[Info] (14400)   Python version: 3.13.2.final.0
[Info] (14400)   Python platform: Windows
[Info] (14400)   Search paths:
[Info] (14400)     c:\Users\Dionisia\.vscode\extensions\ms-python.vscode-pylance-2025.4.1\dist\typeshed-fallback\stdlib
[Info] (14400)     d:\Escritorio\TestGens
[Info] (14400)     d:\Escritorio\TestGens\typings
[Info] (14400)     c:\Users\Dionisia\.vscode\extensions\ms-python.vscode-pylance-2025.4.1\dist\typeshed-fallback\stubs\...
[Info] (14400)     c:\Users\Dionisia\.vscode\extensions\ms-python.vscode-pylance-2025.4.1\dist\bundled\stubs
[Info] (14400)     c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\DLLs
[Info] (14400)     c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\Lib
[Info] (14400)     c:\Users\Dionisia\AppData\Local\Programs\Python\Python313
[Info] (14400)     c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\Lib\site-packages
[Info] (14400) Adding fs watcher for library directories:
 file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib
file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313
[Info] (14400) Adding fs watcher for directories:
 file:///d%3A/Escritorio/TestGens
[Log] (14400) Searching for source files
[Info] (14400) Found 1 source file
[Log] (14400) Adding fs watcher for files:
 file:///d%3A/Escritorio/TestGens/requirements.txt
file:///d%3A/Escritorio/TestGens/pyproject.toml
[Info] (14400) Execution environment: 3.13.2 (global)
[Info] (14400)   Extra paths:
[Info] (14400)     (none)
[Info] (14400)   Python version: 3.13.2.final.0
[Info] (14400)   Python platform: Windows
[Info] (14400)   Search paths:
[Info] (14400)     c:\Users\Dionisia\.vscode\extensions\ms-python.vscode-pylance-2025.4.1\dist\typeshed-fallback\stdlib
[Info] (14400)     \<default workspace root>
[Info] (14400)     \<default workspace root>\typings
[Info] (14400)     c:\Users\Dionisia\.vscode\extensions\ms-python.vscode-pylance-2025.4.1\dist\typeshed-fallback\stubs\...
[Info] (14400)     c:\Users\Dionisia\.vscode\extensions\ms-python.vscode-pylance-2025.4.1\dist\bundled\stubs
[Info] (14400)     c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\DLLs
[Info] (14400)     c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\Lib
[Info] (14400)     c:\Users\Dionisia\AppData\Local\Programs\Python\Python313
[Info] (14400)     c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\Lib\site-packages
[Info] (14400) Adding fs watcher for library directories:
 file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib
file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313
[Info] (14400) Adding fs watcher for directories:
 file:///%3Cdefault%20workspace%20root%3E
[Log] (14400) Searching for source files
[Error] (14400) File or directory "\<default workspace root>" does not exist.
[Info] (14400) No source files found.
[Log] (14400) Adding fs watcher for files:
 file:///%3Cdefault%20workspace%20root%3E/requirements.txt
file:///%3Cdefault%20workspace%20root%3E/pyproject.toml
[Log] (14400) Server settings returned for workspace: file:///d%3A/Escritorio/TestGens: {
  "autoSearchPaths": true,
  "disableLanguageServices": false,
  "openFilesOnly": true,
  "useLibraryCodeForTypes": true,
  "watchForSourceChanges": true,
  "watchForLibraryChanges": true,
  "watchForConfigChanges": true,
  "typeCheckingMode": "strict",
  "diagnosticSeverityOverrides": {
    "reportShadowedImports": "warning"
  },
  "diagnosticBooleanOverrides": {
    "strictListInference": false,
    "strictDictionaryInference": false,
    "strictSetInference": false,
    "analyzeUnannotatedFunctions": true,
    "strictParameterNoneValue": true,
    "enableTypeIgnoreComments": true,
    "deprecateTypingAliases": false,
    "enableReachabilityAnalysis": false,
    "enableExperimentalFeatures": false,
    "disableBytesTypePromotions": false
  },
  "logLevel": "log",
  "autoImportCompletions": false,
  "indexing": true,
  "completeFunctionParens": false,
  "enableExtractCodeAction": true,
  "indexOptions": {
    "packageDepths": [
      [
        "sklearn",
        2,
        false
      ],
      [
        "matplotlib",
        2,
        false
      ],
      [
        "scipy",
        2,
        false
      ],
      [
        "django",
        2,
        false
      ],
      [
        "flask",
        2,
        false
      ],
      [
        "fastapi",
        2,
        false
      ]
    ],
    "regenerateStdLibIndices": false,
    "localStorageUri": {
      "_key": "d:\\escritorio\\testgens",
      "_filePath": "d:\\Escritorio\\TestGens",
      "_query": "",
      "_fragment": "",
      "_originalString": "file:///d%3A/Escritorio/TestGens",
      "_isCaseSensitive": false,
      "_formattedString": "file:///d%3A/Escritorio/TestGens",
      "_normalizedPath": "d:/Escritorio/TestGens"
    },
    "includeAliasesFromUserFiles": false,
    "userFileIndexingLimit": 2000
  },
  "variableInlayTypeHints": false,
  "callArgumentNameInlayHints": "off",
  "functionReturnInlayTypeHints": false,
  "pytestParametersInlayTypeHints": false,
  "importFormat": "absolute",
  "includeFileSpecs": [],
  "excludeFileSpecs": [],
  "ignoreFileSpecs": [],
  "formatOnType": true,
  "taskListTokens": [],
  "enablePytestSupport": true,
  "gotoDefinitionInStringLiteral": true,
  "remapDiagnostics": false,
  "intelliCodeEnabled": true,
  "aiCodeActions": {
    "implementAbstractClasses": true
  },
  "generateWithTypeAnnotation": true,
  "languageServerMode": "default",
  "enableEditableInstalls": false,
  "enablePerfTelemetry": false,
  "enableCompactContext": false,
  "pythonPath": {
    "_key": "c:\\users\\dionisia\\appdata\\local\\programs\\python\\python313\\python.exe",
    "_filePath": "C:\\Users\\Dionisia\\AppData\\Local\\Programs\\Python\\Python313\\python.exe",
    "_query": "",
    "_fragment": "",
    "_isCaseSensitive": false
  },
  "pythonEnvironmentName": "3.13.2 (global)",
  "disableTaggedHints": false,
  "fixAll": [],
  "showOnlyDirectDependenciesInAutoImport": false,
  "extraCommitChars": false,
  "autoFormatStrings": false,
  "autoIndent": true,
  "autoSplitStrings": true,
  "supportRestructuredText": true,
  "cacheLSPData": false,
  "reportExtraTelemetry": false,
  "supportDocstringTemplate": false,
  "displayEnglishDiagnostics": false,
  "diagnosticsSource": "pylance",
  "pyrightVersion": "",
  "nodeExecutable": "",
  "editorIndentConfig": {
    "defaultTabSequence": "    ",
    "tabSize": 4
  }
}
[Log] (14400) Server settings returned for workspace: undefined: {
  "autoSearchPaths": true,
  "disableLanguageServices": false,
  "openFilesOnly": true,
  "useLibraryCodeForTypes": true,
  "watchForSourceChanges": true,
  "watchForLibraryChanges": true,
  "watchForConfigChanges": true,
  "typeCheckingMode": "strict",
  "diagnosticSeverityOverrides": {
    "reportShadowedImports": "warning"
  },
  "diagnosticBooleanOverrides": {
    "strictListInference": false,
    "strictDictionaryInference": false,
    "strictSetInference": false,
    "analyzeUnannotatedFunctions": true,
    "strictParameterNoneValue": true,
    "enableTypeIgnoreComments": true,
    "deprecateTypingAliases": false,
    "enableReachabilityAnalysis": false,
    "enableExperimentalFeatures": false,
    "disableBytesTypePromotions": false
  },
  "logLevel": "log",
  "autoImportCompletions": false,
  "indexing": false,
  "completeFunctionParens": false,
  "enableExtractCodeAction": true,
  "indexOptions": {
    "packageDepths": [
      [
        "sklearn",
        2,
        false
      ],
      [
        "matplotlib",
        2,
        false
      ],
      [
        "scipy",
        2,
        false
      ],
      [
        "django",
        2,
        false
      ],
      [
        "flask",
        2,
        false
      ],
      [
        "fastapi",
        2,
        false
      ]
    ],
    "regenerateStdLibIndices": false,
    "includeAliasesFromUserFiles": false,
    "userFileIndexingLimit": 2000
  },
  "variableInlayTypeHints": false,
  "callArgumentNameInlayHints": "off",
  "functionReturnInlayTypeHints": false,
  "pytestParametersInlayTypeHints": false,
  "importFormat": "absolute",
  "includeFileSpecs": [],
  "excludeFileSpecs": [],
  "ignoreFileSpecs": [],
  "formatOnType": true,
  "taskListTokens": [],
  "enablePytestSupport": true,
  "gotoDefinitionInStringLiteral": true,
  "remapDiagnostics": false,
  "intelliCodeEnabled": true,
  "aiCodeActions": {
    "implementAbstractClasses": true
  },
  "generateWithTypeAnnotation": true,
  "languageServerMode": "default",
  "enableEditableInstalls": false,
  "enablePerfTelemetry": false,
  "enableCompactContext": false,
  "pythonPath": {
    "_key": "c:\\users\\dionisia\\appdata\\local\\programs\\python\\python313\\python.exe",
    "_filePath": "c:\\Users\\Dionisia\\AppData\\Local\\Programs\\Python\\Python313\\python.exe",
    "_query": "",
    "_fragment": "",
    "_originalString": "file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/python.exe",
    "_isCaseSensitive": false,
    "_formattedString": "file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/python.exe"
  },
  "pythonEnvironmentName": "3.13.2 (global)",
  "disableTaggedHints": false,
  "fixAll": [],
  "showOnlyDirectDependenciesInAutoImport": false,
  "extraCommitChars": false,
  "autoFormatStrings": false,
  "autoIndent": true,
  "autoSplitStrings": true,
  "supportRestructuredText": true,
  "cacheLSPData": false,
  "reportExtraTelemetry": false,
  "supportDocstringTemplate": false,
  "displayEnglishDiagnostics": false,
  "diagnosticsSource": "pylance",
  "pyrightVersion": "",
  "nodeExecutable": "",
  "editorIndentConfig": {
    "defaultTabSequence": "    ",
    "tabSize": 4
  }
}
[Log] (14400) BG: Attempting to resolve using local imports: __builtins__
[Log] (14400) BG: [BG(1)] analyzing: file:///d%3A/Escritorio/TestGens/test.py ...
[Log] (14400) BG: [BG(1)]   parsing: file:///d%3A/Escritorio/TestGens/test.py (316ms)
[Log] (14400) BG: [BG(1)]   parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 3ms] (171ms)
[Log] (14400) BG: [BG(1)]   binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/builtins.pyi (22ms)
[Log] (14400) BG: [BG(1)]   binding: file:///d%3A/Escritorio/TestGens/test.py (1ms)
[Log] (14400) BG: [BG(1)]   checking: file:///d%3A/Escritorio/TestGens/test.py ...
[Log] (14400) BG: [BG(1)]     parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/typing.pyi [fs read 2ms] (30ms)
[Log] (14400) BG: [BG(1)]     binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/typing.pyi (14ms)
[Log] (14400) BG: [BG(1)]     parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/types.pyi [fs read 2ms] (13ms)
[Log] (14400) BG: [BG(1)]     binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/types.pyi (4ms)
[Log] (14400) BG: [BG(1)]     parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 1ms] (12ms)
[Log] (14400) BG: [BG(1)]     binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (4ms)
[Log] (14400) BG: [BG(1)]     parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/abc.pyi [fs read 1ms] (4ms)
[Log] (14400) BG: [BG(1)]     binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/abc.pyi (1ms)
[Log] (14400) BG: [BG(1)]     parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/collections/abc.pyi [fs read 1ms] (1ms)
[Log] (14400) BG: [BG(1)]     binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/collections/abc.pyi ...
[Log] (14400) BG: [BG(1)]       parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_collections_abc.pyi [fs read 2ms] (4ms)
[Log] (14400) BG: [BG(1)]       binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_collections_abc.pyi (2ms)
[Log] (14400) BG: [BG(1)]     binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/collections/abc.pyi (6ms)
[Log] (14400) BG: [BG(1)]     parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 2ms] (17ms)
[Log] (14400) BG: [BG(1)]     binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/typing_extensions.pyi (3ms)
[Log] (14400) BG: [BG(1)]   checking: file:///d%3A/Escritorio/TestGens/test.py (371ms)
[Info] (14400) BG: Heap stats: total_memory_size=14250MB, total_free_size=7197MB, total_heap_size=81MB, used_heap_size=68MB, cross_worker_used_heap_size=68MB, total_physical_size=81MB, total_available_size=1991MB, heap_size_limit=2060MB
[Log] (14400) BG: [BG(1)] analyzing: file:///d%3A/Escritorio/TestGens/test.py (882ms)
[Log] (14400) BG: [BG(1)] indexing: file:///d%3A/Escritorio/TestGens/test.py [found 2] (2ms)
[Log] (14400) BG: Workspace indexing done: file:///d%3A/Escritorio/TestGens/test.py
[Log] (14400) Attempting to resolve using local imports: __builtins__
[Log] (14400) [FG] parsing: file:///d%3A/Escritorio/TestGens/test.py (149ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 3ms] (120ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/builtins.pyi (30ms)
[Log] (14400) [FG] binding: file:///d%3A/Escritorio/TestGens/test.py (1ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/cacheprovider.py [fs read 1ms] (41ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/cacheprovider.py (5ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/typing.pyi [fs read 2ms] (44ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/typing.pyi (54ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/types.pyi [fs read 9ms] (54ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/types.pyi (19ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 6ms] (22ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (6ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/abc.pyi [fs read 2ms] (8ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/abc.pyi (3ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/dataclasses.pyi [fs read 2ms] (47ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/dataclasses.pyi (2ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/enum.pyi [fs read 2ms] (17ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/enum.pyi (7ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 2ms] (20ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/typing_extensions.pyi (5ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/collections/abc.pyi [fs read 3ms] (3ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/collections/abc.pyi ...
[Log] (14400) [FG]   parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_collections_abc.pyi [fs read 3ms] (9ms)
[Log] (14400) [FG]   binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_collections_abc.pyi (4ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/collections/abc.pyi (16ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/fixtures.py [fs read 3ms] (98ms)
[Info] (14400) Could not import 'exceptiongroup' in file 'c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\Lib\site-packages\_pytest\fixtures.py'
[Info] (14400)   Looking in stubPath 'file:///d%3A/Escritorio/TestGens/typings'
[Info] (14400)   Attempting to resolve stub package using root path 'file:///d%3A/Escritorio/TestGens/typings'
[Info] (14400)   Attempting to resolve using root path 'file:///d%3A/Escritorio/TestGens/typings'
[Info] (14400)   Looking in root directory of execution environment 'file:///d%3A/Escritorio/TestGens'
[Info] (14400)   Attempting to resolve stub package using root path 'file:///d%3A/Escritorio/TestGens'
[Info] (14400)   Attempting to resolve using root path 'file:///d%3A/Escritorio/TestGens'
[Info] (14400)   Looking for typeshed stdlib path
[Info] (14400)   Looking for typeshed stdlib path
[Info] (14400)   Attempting to resolve using root path 'file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib'
[Info] (14400)   Typeshed path not found
[Info] (14400)   Finding python search paths
[Info] (14400)   Executing interpreter: 'c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\python.exe'
[Info] (14400)   Skipping 'c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\python313.zip' because it is not a valid directory
[Info] (14400)   Received 4 paths from interpreter
[Info] (14400)     file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/DLLs
[Info] (14400)     file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib
[Info] (14400)     file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313
[Info] (14400)     file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages
[Info] (14400)   Looking in python search path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/DLLs'
[Info] (14400)   Attempting to resolve stub package using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/DLLs'
[Info] (14400)   Attempting to resolve using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/DLLs'
[Info] (14400)   Looking in python search path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib'
[Info] (14400)   Attempting to resolve stub package using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib'
[Info] (14400)   Attempting to resolve using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib'
[Info] (14400)   Looking in python search path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313'
[Info] (14400)   Attempting to resolve stub package using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313'
[Info] (14400)   Attempting to resolve using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313'
[Info] (14400)   Looking in python search path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages'
[Info] (14400)   Attempting to resolve stub package using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages'
[Info] (14400)   Attempting to resolve using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages'
[Info] (14400)   Looking in bundled stubs path 'file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/bundled/stubs'
[Info] (14400)   Attempting to resolve stub package using root path 'file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/bundled/stubs'
[Info] (14400)   Attempting to resolve using root path 'file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/bundled/stubs'
[Info] (14400)   Looking for typeshed third-party path
[Info] (14400)   Looking for typeshed stubs path
[Info] (14400)   Typeshed path not found
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/fixtures.py (45ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/config/__init__.py [fs read 3ms] (107ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/config/__init__.py (19ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/capture.py [fs read 2ms] (50ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/capture.py (15ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/io.pyi [fs read 2ms] (11ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/io.pyi (1ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_io.pyi [fs read 2ms] (7ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_io.pyi (6ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/contextlib.pyi [fs read 2ms] (7ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/contextlib.pyi (2ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/doctest.py [fs read 2ms] (18ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/doctest.py (8ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/junitxml.py [fs read 1ms] (19ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/junitxml.py (8ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/legacypath.py [fs read 4ms] (16ms)
[Info] (14400) Could not resolve source for 'pexpect' in file 'c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\Lib\site-packages\_pytest\legacypath.py'
[Info] (14400)   Looking in root directory of execution environment 'file:///d%3A/Escritorio/TestGens'
[Info] (14400)   Attempting to resolve using root path 'file:///d%3A/Escritorio/TestGens'
[Info] (14400)   Finding python search paths
[Info] (14400)   Executing interpreter: 'c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\python.exe'
[Info] (14400)   Skipping 'c:\Users\Dionisia\AppData\Local\Programs\Python\Python313\python313.zip' because it is not a valid directory
[Info] (14400)   Received 4 paths from interpreter
[Info] (14400)     file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/DLLs
[Info] (14400)     file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib
[Info] (14400)     file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313
[Info] (14400)     file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages
[Info] (14400)   Looking in python search path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/DLLs'
[Info] (14400)   Attempting to resolve using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/DLLs'
[Info] (14400)   Looking in python search path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib'
[Info] (14400)   Attempting to resolve using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib'
[Info] (14400)   Looking in python search path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313'
[Info] (14400)   Attempting to resolve using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313'
[Info] (14400)   Looking in python search path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages'
[Info] (14400)   Attempting to resolve using root path 'file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages'
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/legacypath.py (3ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/logging.py [fs read 0ms] (12ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/logging.py (4ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/re.pyi [fs read 1ms] (5ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/re.pyi (2ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/monkeypatch.py [fs read 1ms] (8ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/monkeypatch.py (5ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/recwarn.py [fs read 1ms] (4ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/recwarn.py (4ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/warnings.pyi [fs read 2ms] (3ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/warnings.pyi (1ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/tmpdir.py [fs read 0ms] (5ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/_pytest/tmpdir.py (1ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/pytest_asyncio/plugin.py [fs read 1ms] (25ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/pytest_asyncio/plugin.py (7ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_ast.pyi [fs read 2ms] (4ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/_ast.pyi (1ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/ast.pyi [fs read 2ms] (28ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/ast.pyi (19ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/os/__init__.pyi [fs read 2ms] (16ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/os/__init__.pyi (11ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/__future__.pyi [fs read 2ms] (3ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/__future__.pyi (1ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/asyncio/__init__.pyi [fs read 2ms] (12ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/asyncio/__init__.pyi (1ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/contextvars.pyi [fs read 2ms] (2ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/contextvars.pyi (0ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/functools.pyi [fs read 1ms] (6ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/functools.pyi (3ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/inspect.pyi [fs read 1ms] (10ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/inspect.pyi (5ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/socket.pyi [fs read 4ms] (14ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/socket.pyi (3ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/textwrap.pyi [fs read 1ms] (3ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/.vscode/extensions/ms-python.vscode-pylance-2025.4.1/dist/typeshed-fallback/stdlib/textwrap.pyi (0ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/pluggy/__init__.py [fs read 1ms] (1ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/pluggy/__init__.py (1ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/pytest/__init__.py [fs read 1ms] (8ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/pytest/__init__.py (0ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/pytest_asyncio/_version.py [fs read 0ms] (1ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/pytest_asyncio/_version.py (0ms)
[Log] (14400) [FG] parsing: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/pytest_asyncio/__init__.py [fs read 0ms] (3ms)
[Log] (14400) [FG] binding: file:///c%3A/Users/Dionisia/AppData/Local/Programs/Python/Python313/Lib/site-packages/pytest_asyncio/__init__.py (0ms)
```

</details>

**Comment by erictraut** [comment]
This is effectively a duplicate of #10278.

**Comment by erictraut** [comment]
I'll mark this as a separate enhancement request, but my conclusion is the same as the other one. I'm going to reject this enhancement request for now because I think this is pretty esoteric functionality, and I'm not convinced it's worth adding the extra complexity to pyright's code.

I'm willing to revisit this decision if and when I receive signal from enough other pyright users that this is an important use case to support (in the form of upvotes on this issue). In the meantime, you can provide an explicit return type annotation if you'd like.

**Comment by dev-ankit** [comment]
Adding a concrete use case, since the close comment asked for signal.

I'm building an Effect-TS-style typed effect system for Python (`Effect[A, E, R]` with tracked errors and dependencies), where the entire user-facing API is generator sequencing — the Python analogue of Effect-TS's `yield*`:

```python
@fn("Users.find_by_id")
def find_by_id(user_id: str):
    db = yield from service(Database)          # adds Database to R
    row = db.find(user_id)
    if row is None:
        return (yield from NotFound(user_id))  # adds NotFound to E
    return row

# inferred: (user_id: str) -> Effect[dict[str, str], NotFound, Database]
```

Pyright's inference handles this beautifully — the yield type accumulates the E/R unions across `yield from`, and the decorator solves `Effect[A, E, R]` exactly. The send type is the one hole: it is always `Unknown` for `yield from`-only generators, so under strict mode every such function's def site trips `reportUnknownParameterType` ("Return type `Generator[..., Unknown, ...]` is partially unknown"). The choices for users are disabling that rule wherever effects are defined, or annotating every generator with the full `Generator[Effect[object, E, R], object, A]` spelling — which gives up the inference that makes the pattern viable.

For comparison, TypeScript propagates `TNext` through `yield*` delegation, which is why the same model in Effect-TS runs under `strict` with no annotations and no rule carve-outs.

This would also be a natural extension of the improvement from #5181 (send type inferred as `Any` when provably unused): when every yield in a generator is a `yield from`, sends are forwarded to the delegates, so the join of the delegates' send types is a sound — and precise — inference.
